src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -170,7 +170,9 @@ trait TypeDiagnostics {
 
     def patternMessage    = "pattern " + tree.tpe.finalResultType + valueParamsString(tree.tpe)
     def exprMessage       = "expression of type " + tree.tpe
-    def overloadedMessage = s"overloaded method $sym with alternatives:\n" + alternativesString(tree)
+    def overloadedMessage =
+      if (sym.isConstructor) s"multiple constructors for ${sym.owner.decodedName}${sym.idString} with alternatives:\n${alternativesString(tree)}"
+      else s"overloaded method $sym with alternatives:\n${alternativesString(tree)}"
```

Comment:
`${sym.name}`? I just checked and this says "overloaded method value foobar with alternatives:` which I suppose is because we don't slap the `METHOD` flag on overloaded method symbols.

Also method values are on their way out, I thought.

Reply:
`valued method`?